### DISCUSSION TITLE
Slashcommand to search for translation

### DISCUSSION THREAD

**Discussion by kolomu** [arnoudkooi/SN-Utils#455] [General]
Good day :)

It would be convenient to have a slash command to search in multiple tables for a translation. 

I have created the following script for this:

```javascript
var stringToEncode = `
var searchString = "ungültig";
var results = [];

var tableMap = {
    "sys_ui_message": ["messageLIKE"],
    "sys_translated": ["labelLIKE", "valueLIKE"],
    "sys_translated_text": ["valueLIKE"]
};

Object.keys(tableMap).forEach(function(tableName) {
    for(var i = 0; i < tableMap[tableName].length; i++) {
        var gr = new GlideRecord(tableName);
        gr.addEncodedQuery(tableMap[tableName][i] + searchString);
        gr.query();
        if(gr.next()) {
            results.push(tableName + " found " + gr.getUniqueValue() + " " + gr.getDisplayValue());
        }
    }
});

gs.info("------------------------- RESULTS -------------------------");
gs.info(results.length == 0 ? "Found nothing" : results.join("\\n"));
gs.info("----------------------- END RESULTS -----------------------");
`;
```
Then I use the encodeURIComponent function to get an encoded string which I can paste for creating a slash command.

```
%0Avar%20searchString%20%3D%20%22ung%C3%BCltig%22%3B%0Avar%20results%20%3D%20%5B%5D%3B%0A%0Avar%20tableMap%20%3D%20%7B%0A%20%20%20%20%22sys_ui_message%22%3A%20%5B%22messageLIKE%22%5D%2C%0A%20%20%20%20%22sys_translated%22%3A%20%5B%22labelLIKE%22%2C%20%22valueLIKE%22%5D%2C%0A%20%20%20%20%22sys_translated_text%22%3A%20%5B%22valueLIKE%22%5D%0A%7D%3B%0A%0AObject.keys(tableMap).forEach(function(tableName)%20%7B%0A%20%20%20%20for(var%20i%20%3D%200%3B%20i%20%3C%20tableMap%5BtableName%5D.length%3B%20i%2B%2B)%20%7B%0A%20%20%20%20%20%20%20%20var%20gr%20%3D%20new%20GlideRecord(tableName)%3B%0A%20%20%20%20%20%20%20%20gr.addEncodedQuery(tableMap%5BtableName%5D%5Bi%5D%20%2B%20searchString)%3B%0A%20%20%20%20%20%20%20%20gr.query()%3B%0A%20%20%20%20%20%20%20%20if(gr.next())%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20results.push(tableName%20%2B%20%22%20found%20%22%20%2B%20gr.getUniqueValue()%20%2B%20%22%20%22%20%2B%20gr.getDisplayValue())%3B%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%7D)%3B%0A%0Ags.info(%22-------------------------%20RESULTS%20-------------------------%22)%3B%0Ags.info(results.length%20%3D%3D%200%20%3F%20%22Found%20nothing%22%20%3A%20results.join(%22%5Cn%22))%3B%0Ags.info(%22-----------------------%20END%20RESULTS%20-----------------------%22)%3B%0A
```

When I try to insert it as custom slash command the \n get interpreted as newline which result in a wrong script:

<img width="1079" alt="Screenshot 2023-12-09 at 11 41 46" src="https://github.com/arnoudkooi/ServiceNow-Utils/assets/12582455/ac78319e-98c8-493a-a98a-b9daf489ac8f">

I tried escaping the newline but it seems that doesn't work. Do you have any ideas how to proceed from here on? 

Kind regards

**Comment by arnoudkooi** [comment]
No idea, for me your snippet worked 
![Google Chrome20231209 - 1256](https://github.com/arnoudkooi/ServiceNow-Utils/assets/840781/3f63aa4e-f206-463f-9269-7990afae1a9a)

**Reply by kolomu** [reply]
Thanks I tried it again and now it works. Don't know what the issue was.

**Comment by kolomu** [comment]
Is there a possibility to add a slash command which loads a large custom script in the ServiceNow background script text field? My issue is that the current slash command is too long to be pasted as URI with [sys.scripts.do](http://sys.scripts.do/). It should be independent of instances like calling a script include. Maybe using the local storage of the pc? I would appreciate if you have a hint for me. Thanks in advance and have a nice day :).
